src/allowlist/AllowList.sol

```diff
@@ -45,6 +49,11 @@ contract AllowList is IAllowListV2, Ownable2StepUpgradeable {
     }
 
     function isAddressAllowedForFund(address addr, string calldata fundSymbol) external view returns (bool) {
+        // First check if address has contract permissions
+        if (EntityId.unwrap(addressEntityIds[addr]) == 0) {
```

Comment:
nit: i did a double take at first, as random addresses will be unset and entity will be 0 also. maybe a different comment would make more readable. or maybe reordering to checking the entity != 0 case first.